tldw_Server_API/tests/AuthNZ_Federation/test_oidc_login_flow.py

```
@@ -0,0 +1,733 @@
+    provider = create_response.json()
+
+    login_response = federation_client.get(
+        "/api/v1/auth/federation/corp/login",
+        follow_redirects=False,
+    )
+    assert login_response.status_code == 307, login_response.text
+    login_query = parse_qs(urlparse(login_response.headers["location"]).query)
+    state = login_query["state"][0]
+
+    async def _fake_exchange(
+        self,
```

Ruff (0.15.6)

[warning] Unused function argument: `self`

(ARG001)